torchgeo/datasets/mdas.py

```diff
@@ -0,0 +1,347 @@
+        7201: 1,
+        7202: 2,
+        7203: 3,
+        7204: 4,
+        7205: 5,
+        7206: 6,
+        7207: 7,
+        7208: 8,
+        7209: 9,
+        7210: 10,
+        7211: 11,
+        7212: 12,
+        7213: 13,
+        7214: 14,
+        7215: 15,
+        7217: 16,
+        7218: 17,
+        7219: 18,
+    }
+
+    cmap = ListedColormap([plt.cm.tab20(i) for i in range(20)])  # type: ignore[attr-defined]
+
+    ds_root_name = 'Augsburg_data_4_publication'
+
+    zipfilename = f'{ds_root_name}.zip'
+
+    valid_subareas = ('sub_area_1', 'sub_area_2', 'sub_area_3')
+
+    url = 'https://huggingface.co/datasets/torchgeo/mdas/resolve/860226b74269f1cf1bed8ea3c03f571ae701144c/Augsburg_data_4_publication.zip'
+
+    md5 = '7b63c26e3717cb52c6ba47d215f18d5b'
+
+    enmap_rgb_band_idx: ClassVar[list[int]] = [43, 28, 10]
+    sentinel_2_rgb_band_idx: ClassVar[list[int]] = [3, 2, 1]
+    hyspex_rgb_band_idx: ClassVar[list[int]] = [100, 50, 10]
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        subareas: list[str] = ['sub_area_1'],
+        modalities: list[str] = ['3K_RGB', 'HySpex', 'Sentinel_2'],
+        transforms: Callable[[dict[str, Any]], dict[str, Any]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MDAS dataset instance.
+
+        Args:
+            root: Root directory where the dataset should be stored.
+            subareas: The subareas to load. Options are 'sub_area_1', 'sub_area_2', 'sub_area_3'.
+            modalities: The modalities to load. Options are '3K_DSM', '3K_RGB', 'HySpex', 'EeteS_EnMAP_10m', 'EeteS_EnMAP_30m', 'EeteS_Sentinel_2_10m', 'Sentinel-2', 'Sentinel-1', 'OSM_label'.
+            transforms: A function/transform that takes in a dictionary and returns a transformed version.
+            download: if True, download dataset and store it in the root directory
+            checksum: If True, check the integrity of the dataset after download.
+
+        Raises:
+            AssertionError: If the subareas or modalities are not valid.
+            DatasetNotFoundError: If the dataset is not found.
+        """
+        self.root = root
+        self.download = download
+        assert all(
+            sub in self.valid_subareas for sub in subareas
+        ), f'Subareas must be one of {self.valid_subareas}'
+        self.subareas = subareas
+        assert all(
+            mod in self.valid_modalities for mod in modalities
+        ), f'Modalities must be one of {self.valid_modalities}'
+        self.modalities = modalities
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._verify()
+        self.files = self._load_files()
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset."""
+        return len(self.files)
+
+    def _load_files(self) -> list[dict[str, str]]:
+        """Return the paths of the files in the dataset."""
```

Comment:
Should technically have a `Returns:` section in these docstrings. I know it's stupid when you can describe it in 1 line though. Ruff doesn't currently check for this but I would like it to someday. Can either fix it now or make me fix it in the future.